### PR TITLE
Split page layout into differents blade components

### PR DESCRIPTION
**Pull request by Shidersz**
Splits the **page.blade.php** layout into different components (_top navbar, left sidebar, right sidebar, footer, etc..._), creating multiple blade templates for each one of these and then include they on the page layout in order to make it cleaner and easy to maintain. As an example, this is how the **page.blade.php** file will look after these changes:

```
@extends('adminlte::master')

@section('adminlte_css')
    @stack('css')
    @yield('css')
@stop

@section('classes_body', $adminlte->getBodyClasses())

@section('body_data', $adminlte->getBodyData())

@section('body')
    <div class="wrapper">

        {{-- Top Navbar --}}
        @if(config('adminlte.layout_topnav') || View::getSection('layout_topnav'))
            @include('adminlte::partials.navbar.navbar-layout-topnav')
        @else
            @include('adminlte::partials.navbar.navbar')
        @endif

        {{-- Left Main Sidebar --}}
        @if(!config('adminlte.layout_topnav') && !View::getSection('layout_topnav'))
            @include('adminlte::partials.sidebar.left-sidebar')
        @endif

        {{-- Content Wrapper --}}
        <div class="content-wrapper">
            @if(config('adminlte.layout_topnav') || View::getSection('layout_topnav'))
                <div class="container">
            @endif

            {{-- Content Header --}}
            <div class="content-header">
                <div class="{{config('adminlte.classes_content_header', 'container-fluid')}}">
                    @yield('content_header')
                </div>
            </div>

            {{-- Main Content --}}
            <div class="content">
                <div class="{{config('adminlte.classes_content', 'container-fluid')}}">
                    @yield('content')
                </div>
            </div>

            @if(config('adminlte.layout_topnav') || View::getSection('layout_topnav'))
                </div>
            @endif
        </div>

        {{-- Footer --}}
        @hasSection('footer')
            @include('adminlte::partials.footer.footer')
        @endif

        {{-- Right Control Sidebar --}}
        @if(config('adminlte.right_sidebar'))
            @include('adminlte::partials.sidebar.right-sidebar')
        @endif

    </div>
@stop

@section('adminlte_js')
    <script src="{{ asset('vendor/adminlte/dist/js/adminlte.min.js') }}"></script>
    @stack('js')
    @yield('js')
@stop
```
Where, for example, **adminlte::partials.navbar.navbar** will be:

```
<nav class="main-header navbar
    {{ config('adminlte.classes_topnav_nav', 'navbar-expand-md') }}
    {{ config('adminlte.classes_topnav', 'navbar-white navbar-light') }}">

    {{-- Navbar left links --}}
    <ul class="navbar-nav">
        {{-- Left sidebar toggler link --}}
        @include('adminlte::partials.navbar.left-sidebar-link')

        @each('adminlte::partials.menuitems.menu-item-top-nav-left', $adminlte->menu(), 'item')
        @yield('content_top_nav_left')
    </ul>

    {{-- Navbar right links --}}
    <ul class="navbar-nav ml-auto">
        @yield('content_top_nav_right')
        @each('adminlte::partials.menuitems.menu-item-top-nav-right', $adminlte->menu(), 'item')

        {{-- User menu links --}}
        @if(Auth::user())
            @if(config('adminlte.usermenu_enabled'))
                @include('adminlte::partials.navbar.dropdown-user-menu')
            @else
                @include('adminlte::partials.navbar.logout-link')
            @endif
        @endif

        {{-- Right sidebar toggler link --}}
        @if(config('adminlte.right_sidebar'))
            @include('adminlte::partials.navbar.right-sidebar-link')
        @endif
    </ul>

</nav>
```
And  **adminlte::partials.navbar.logout-link** will be:

```
@php( $logout_url = View::getSection('logout_url') ?? config('adminlte.logout_url', 'logout') )

@if (config('adminlte.use_route_url', false))
    @php( $logout_url = $logout_url ? route($logout_url) : '' )
@else
    @php( $logout_url = $logout_url ? url($logout_url) : '' )
@endif

<li class="nav-item">
    <a class="nav-link" href="#" onclick="event.preventDefault(); document.getElementById('logout-form').submit();">
        <i class="fa fa-fw fa-power-off"></i>
        {{ __('adminlte::adminlte.log_out') }}
    </a>
    <form id="logout-form" action="{{ $logout_url }}" method="POST" style="display: none;">
        @if(config('adminlte.logout_method'))
            {{ method_field(config('adminlte.logout_method')) }}
        @endif
        {{ csrf_field() }}
    </form>
</li>
```

Also, this is how the structure of the **resources/views** folder will look like:

```
resources/views/
├── login.blade.php
├── master.blade.php
├── page.blade.php
├── partials
│   ├── common
│   │   ├── brand-logo-xl.blade.php
│   │   └── brand-logo-xs.blade.php
│   ├── footer
│   │   └── footer.blade.php
│   ├── menuitems
│   │   ├── menu-item.blade.php
│   │   ├── menu-item-sub-top-nav.blade.php
│   │   ├── menu-item-top-nav.blade.php
│   │   ├── menu-item-top-nav-left.blade.php
│   │   ├── menu-item-top-nav-right.blade.php
│   │   └── menu-item-top-nav-user.blade.php
│   ├── navbar
│   │   ├── dropdown-user-menu.blade.php
│   │   ├── left-sidebar-link.blade.php
│   │   ├── logout-link.blade.php
│   │   ├── navbar.blade.php
│   │   ├── navbar-layout-topnav.blade.php
│   │   └── right-sidebar-link.blade.php
│   └── sidebar
│       ├── left-sidebar.blade.php
│       └── right-sidebar.blade.php
├── passwords
│   ├── confirm.blade.php
│   ├── email.blade.php
│   └── reset.blade.php
├── plugins.blade.php
├── register.blade.php
└── verify.blade.php
```
I'm planning, still, to give some review to these changes and maybe add more commits.

### CODE REVIEW

**Comment by andcarpi** [comment]
I would suggest a note on the readme, because that can be a breaking change for people that published and modified the views.

**Comment by Shidersz** [comment]
@andcarpi Thanks for the feedback, I'm going to add that into my list of pending reviews.

**Comment by REJack** [comment]
There should be no breaking change if the users published them, in this case the user use their published files only. Only if they republish the main view it could cause some breaks.

**Comment by lacodimizer** [comment]
@REJack I think it will be better to release that as minor release v3.2. What do you mean?

**Comment by Shidersz** [comment]
@lacodimizer @REJack The latest two commits was made in order to pass the tests (I have copy that from other PR that is already merged with _master_). Will they cause a problem on the future, for example, when this PR is going to be merged with master?

**Comment by lacodimizer** [comment]
@Shidersz no, I don't think.

**Comment by Shidersz** [comment]
@andcarpi  @REJack I have updated the **README** file as suggested, do you think it is enough or I'm missing another section where I have to make a note?

**Comment by Shidersz** [comment]
Explanation of the latest 3 commits:

When inspecting the layouts of [AdminLTE V3](https://adminlte.io/themes/v3/) you can see the next `html` structure for the `content-wrapper`:

```
<div class="content-wrapper">
    <div class="content-header">
        <div class="container-fluid">
            <!-- content header -->
        </div>
    </div>
     <div class="content">
        <div class="container-fluid">
            <!-- content -->
        </div>
     </div>
</div>
```
But, when **layout topnav** is used `container-fluid` class is changed to `container` class. So, the changes provides the same behavior. Also, now configuration options `classes_content_header` and `classes_content` will add extra classes to elements `div.content-header` and `div.content` respectively. Finally, README is updated in order to provide information about what elements the extra classes affect.

**Comment by lacodimizer** [comment]
@Shidersz great work! @andcarpi @REJack can you test it too? I will publish the next release v3.2 soon. 💪

File: resources/views/page.blade.php [hunk 1]
```diff
@@ -1,5 +1,11 @@
 @extends('adminlte::master')
 
+@if(config('adminlte.layout_topnav') || View::getSection('layout_topnav'))
```
Comment (REJack) [inline resources/views/page.blade.php:3]: This should not be directed by layout-topnav, you can use one of the two classes with any layout option enabled.

Comment (Shidersz) [reply resources/views/page.blade.php:3]: Ok, I'm going to give a review. That change was based on the examples shown by the AdminLTE site. So, you say any of the classes "container" or "container-fluid" can be used independent of the layout, but is one of they always required?

Comment (Shidersz) [reply resources/views/page.blade.php:3]: I have made a new commit that take this into considerations, now `$def_container_class` will be used only as a default value, when the configuration options are empty. This implies that the config options `classes_content` and `classes_content_header` will now affects the containers of `div.content` and `div.content-header` as previously did. Can you check it now?

**Comment by REJack** [comment]
I did a quick test and it looks good and works, I could not found any error/bug.

**Pull request merged**